distributed/shuffle/_scheduler_plugin.py

```diff
@@ -96,10 +97,44 @@ async def barrier(self, id: ShuffleId, run_id: int, consistent: bool) -> None:
                 stimulus_id=f"p2p-barrier-inconsistent-{time()}",
             )
         msg = {"op": "shuffle_inputs_done", "shuffle_id": id, "run_id": run_id}
-        await self.scheduler.broadcast(
-            msg=msg,
-            workers=list(shuffle.participating_workers),
-        )
+        workers = list(shuffle.participating_workers)
+        no_progress = 0
+        while workers:
+            res = await self.scheduler.broadcast(
+                msg=msg,
+                workers=workers,
+                on_error="return",
+            )
+            before = len(workers)
+            workers = []
+            for w, r in res.items():
+                if r is None:
+                    continue
+                if isinstance(r, OSError):
+                    workers.append(w)
+                else:
+                    raise P2PConsistencyError(
```

Comment:
Why raise a `P2PConsistencyError` here? I think we should just propagate the original error instead.